pkg/plugins/resources/gitlab/mergerequest/utils.go

```diff
@@ -102,26 +100,25 @@ func (g *Gitlab) isRemoteBranchesExist() (bool, error) {
 	foundRemoteSourceBranch := false
 	foundRemoteTargetBranch := false
 	page := 0
+	const perPage = 30
 	for {
-		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		ctx, cancel := context.WithTimeout(ctx, gitlabRequestTimeout)
 		defer cancel()
-		remoteBranches, resp, err := g.client.Git.ListBranches(
-			ctx,
-			strings.Join([]string{owner, repository}, "/"),
-			scm.ListOptions{
-				URL:  g.spec.URL,
-				Page: page,
-				Size: 30,
+		remoteBranches, resp, err := g.api.Branches.ListBranches(
+			g.getPID(),
+			&gitlabapi.ListBranchesOptions{
+				ListOptions: gitlabapi.ListOptions{
+					Page:    page,
+					PerPage: perPage,
+				},
 			},
+			gitlabapi.WithContext(ctx),
 		)
 
-		if err != nil {
-			logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
-			return false, err
-		}
-
-		if resp.Status > 400 {
-			logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
+		if err != nil || resp.StatusCode > 400 {
```

Comment:
Same here, resp and err can't be tested at the same time becaues resp is nil if err is not 

https://gitlab.com/gitlab-org/api/client-go/-/blob/v0.128.0/branches.go#L84

Reply:
Fixed